src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testContactAddressMarkOverrideWithStateAndCountryPicklistsEnabled') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+        System.assertEquals(1, getAddresses().size());
+
+        // Modify the Contact addresses directly, and set override
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.put('MailingCountryCode', 'US');
+        con.is_Address_Override__c = true;
+
+        Test.startTest();
+        update con;
+        Test.stopTest();
+
+        // Verify a new Address record was created
+        List<Address__c> listAddr = getAddresses();
+        System.assertEquals(2, listAddr.size());
+
+        // Populate a Map of AccountId to Address record
+        Map<Id, Address__c> accountIdToAddress = new Map<Id, Address__c>();
+        for(Address__c eachAddress : listAddr) {
+            if(eachAddress.Default_Address__c) {
+              accountIdToAddress.put(eachAddress.Household_Account__c, eachAddress);
+            }
+        }
+
+        // Verify the Account's Billing Address info matches the Default Address record
+        List<Account> retrievedAccounts = Database.query(buildAccountSoql());
+        assertEquals(listAddrT[0], accountIdToAddress.get(retrievedAccounts[0].Id));
+
+        // Verify the Contact's Address was updated as expected and that the Household and Contact Addresses are different.
+        Map<Id, Account> accountIdToAccount = new Map<Id, Account>(retrievedAccounts);
+        List<Contact> listCon = Database.query(buildContactSoql());
+        for (Contact eachContact : listCon) {
+            System.assertEquals(true, eachContact.MailingState.equals('California'));
+            System.assertEquals(true, eachContact.is_Address_Override__c);
+            System.assertNotEquals(null, eachContact.Current_Address__c);
+
+            Account eachContactAccount = accountIdToAccount.get(eachContact.AccountId);
+            System.assertEquals(false, isMatchAddressAccCon(eachContactAccount, eachContact));
+        }
+
+        // Verify the New Address created matches the Contact's Address and not set the Default
+        for (Integer i = 0; i < listAddr.size(); i++) {
+            if(listAddr[i].MailingState__c.equals('California')) {
+                assertEqualAddresses(listAddr[i], listCon[i]);
+                System.assertEquals(false, listAddr[i].Default_Address__c);
+            }
+        }
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing Country and State (with State & Country Picklists enabled) when the Country is empty
+        or the State empty and/or mismatched.
```

Comment:
Since we are testing empty State/Country only, perhaps the test method description should exclude `mismatched` since it is not tested here?